### DISCUSSION TITLE
4000 character limit in tweets

### DISCUSSION THREAD

**Discussion by dnestico** [robertodoering/harpy#855] [General]
So Elon Musk confirmed that Twitter is planning to raise the character limit to 4000 characters so just wanted to open a discussion / share some thoughts before / if it comes to Harpy.

![Screenshot-2022-12-12-at-12 55 53-PM-e1670830003519](https://user-images.githubusercontent.com/3034555/207161738-e125156f-b91e-41b9-8a17-97d7002d9c8d.png)

First of all, this is how 4000 characters would look.... 🤦 🙈 

![FjuQgTwaEAA5vUS](https://user-images.githubusercontent.com/3034555/207161729-68781b90-5e5a-4fe9-95bb-f2b216c0d48d.jpg)

So if Harpy does add support for the 4000 characters please can you make it so that the tweet ends at the normal 280 characters and after 280 characters just put a  **"more..." or "expand..." button that opens in its own tweet page rather than on the normal timeline.** 

I hope the Twitter app does the same for those still using it because this could really get ugly and annoying on the timeline if people start actually going beyond the regular 280 characters.

**Comment by robertodoering** [comment]
Thanks for sharing! Yeah if the character limit is increased to 4000, truncating the text in the timeline to a reasonable amount makes a lot of sense, good idea!

**Reply by dnestico** [reply]
Just a heads up looks like it's gonna launch in February https://twitter.com/PopBase/status/1611959943674105859
